tests/integration/models/tags.spec.coffee

```diff
@@ -0,0 +1,188 @@
+###
+The methods of this file should be imported and used in
+the respective resource tests that support the tag related methods
+eg: application.spec, device.spec
+###
+_ = require('lodash')
+m = require('mochainon')
+{ titleCase } = require('../../util')
+
+getAllByResourceFactory = (model, resourceName) ->
+	(idOrUniqueParam) ->
+		model["getAllBy#{titleCase(resourceName)}"](idOrUniqueParam)
+
+exports.itShouldGetAllTagsByResource = (opts) ->
+	{ model, resourceName, uniquePropertyName } = opts
```

Comment:
Here (and below), you can just inline this destructuring into the function definition, there's no need for the extra `opts` step:

```coffeescript
exports.abc = ({ model, resourceName, ... }) ->
```

Reply:
Ah, I see. Passing in options and then mutating them later (and depending on that process) like this is kind of confusing though. It feels like we should be passing a getter that grabs this, but you explained why that's difficult above, and that does make sense. Still feels awkward and a bit annoying, but I can't see an easy improvement.

For now let's just stick with it, but if you have any ideas to make this clearer in future I'd love to see them, as this all uses a few more clever tricks than I'd like right now.

Reply:
Cool. Totally agree that it's a bit annoying but seemed to be the less complicated (tried the getter & several other approaches). I will keep that in mind for improvement.

Reply:
How about changing `resource` & `setTagResource` to `resourceProvider()` & `setTagResourceProvider()`?
It's almost the same, but I think that making it a method makes it a bit clearer that this is not a fixed property but its something that changes.